radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/ProxiedPeersView.java

```diff
@@ -0,0 +1,122 @@
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.network.p2p;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.radixdlt.environment.RemoteEventDispatcher;
+import com.radixdlt.network.p2p.P2PConfig.ProxyConfig;
+import com.radixdlt.network.p2p.PeerEvent.PeerConnected;
+import com.radixdlt.network.p2p.PeerEvent.PeerDisconnected;
+import com.radixdlt.network.p2p.discovery.ProxiedPeers;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public final class ProxiedPeersView implements PeersView {
+  private final PeerManager peerManager;
+  private final ProxyConfig proxyConfig;
+  private final RemoteEventDispatcher<ProxiedPeers> dispatcher;
+  private Set<RadixNodeUri> peers = Set.of();
+
+  @Inject
+  public ProxiedPeersView(
+      PeerManager peerManager,
+      ProxyConfig proxyConfig,
+      RemoteEventDispatcher<ProxiedPeers> dispatcher) {
```

Comment:
might be just my personal taste, but I prefer to use full names for dispatchers: `proxiedPeersDispatcher`

Reply:
Done